src/formats/notion/convert-to-md.ts

```diff
@@ -235,17 +235,24 @@ function fixEquations(body: HTMLElement) {
 	for (const figEqn of figEqnEls) {
 		const annotation = figEqn.find('annotation');
 		if (!annotation) continue;
-		figEqn.replaceWith(`$$${annotation.textContent}$$`);
+		figEqn.replaceWith(`$$${formatMath(annotation.textContent)}$$`);
 	}
 	// Inline Equations
 	const spanEqnEls = body.findAll('span.notion-text-equation-token');
 	for (const spanEqn of spanEqnEls) {
 		const annotation = spanEqn.find('annotation');
 		if (!annotation) continue;
-		spanEqn.replaceWith(`$${annotation.textContent}$`);
+		spanEqn.replaceWith(`$${formatMath(annotation.textContent, true)}$`);
 	}
 }
 
+function formatMath(math: string | null | undefined, inline: boolean=false): string {
+	// Trim lead/trailing space in LaTeX math experessions.
+	// Removes empty lines, which can cause equations to break.
+	let regex = new RegExp(/^(?:[\s\r\n]|\\\\|\\\s)*(.*?)[\s\r\n\\]*$/, 's');
```

Comment:
I'm not completely familiar with math expressions, so apologies if this is a silly question.

Could we simplify this with something like `math.trim()`, and then possibly a simpler regex replacement afterwards?

What is the `\\\\` matching in this regex?

Reply:
Originally I did use trim plus a regex for the escape sequences; I ended up combining them but I'm happy to change it back, as I can see how that might be clearer.

In LaTeX math: `\\` and `\ ` are the escapes for line-breaks and white-space, respectively.
Depending on the engine, `\\` line-breaks may or may not be parsed — Notion actually

Reply:
In LaTeX math: `\\` and `\ ` are the escapes for line-breaks and white-space, respectively — `\\\\` and `\\\s` in the regex.
Not all LaTeX engines parse `\\` — Notion _does_, but Obsidian MD does not, hence the need to clean up some of the syntax.
It can cause issues in Obsidian (except in a supported environment, such as `cases` or `array`).

Originally I did use trim _plus_ a regex for the escape sequences; I ended up combining them to account for repetition:
The issue with `\ ` is as follows: consider `$f(x)=e^x \ $` which is ok in Notion but
1. this breaks in Obsidian so we `trim` trailing spaces to give `$f(x)=e^x \$`
2. this breaks again due to the escaped `\$`, so remove this to give `$f(x)=e^x $`
3. we're back to the trailing space issue so have to `trim` a second time leaving us with `$f(x)=e^x$`, which is now ok in Obsidian.

Reply:
As a side tangent... arguably code like this is not particularly "good practice" in LaTeX and isn't reliably supported by all engines; however, Notion syntax is much more forgiving than Obsidian so there's no reason a user would need to be vigilant (including myself, as I found out the hard way).
The thing that got me looking into these bugs in the first place was because my Notion workspace contains a lot of math, with some pretty complex code, half of which breaks in the Importer conversion process (things get a lot worse when nested in callouts and bullets).

Reply:
Very helpful, thank you.

> In LaTeX math: \\ and \ are the escapes for line-breaks and white-space, respectively — \\\\ and \\\s in the regex.

This was the key I was missing. Would you mind adding this as a comment below the existing two comment lines?

Reply:
Sure! All done :)